docs/proposals/0845-scheduler-architecture-proposal/interfaces/interface.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+
+	scheduling "sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/types"
+)
+
+// READER NOTE: Currently CycleState is assumed to have appropriate request data rather that making a new object.
+
+// Plugin is the parent type for all the scheduling framework plugins.
+type Plugin interface {
+	Name() string
+}
+
+type Endpoint struct {
+	State EndpointState
+	Score float64
+}
+
+type EndpointState struct {
+	// storage is per Scheduling Cycle, and so has no thread-safe concerns.
+	storage map[string]any
+}
+
+type SchedulingResult struct {
+	results map[string][]Endpoint
+}
```

Comment:
why do we need to wrap a map with a struct?
what are the downsides of returning a map?